rfc/2020/conformance-validation/mayankshah1607/rfc.md

```diff
@@ -0,0 +1,477 @@
+
+
+## Corner Cases
+
+It is possible to have prior knowledge regarding which use cases may need to be configured differently depending on the environment. For E.g, tap requires extra RBAC configurations on GKE. Instead of having the corresponding tests for tap failing, users may be allowed to explicitly mention the environment these tests shall run on. The testing tool may show a warning, providing a link to the docs which provide instructions on how to setup GKE for Linkerd. This would have a positive impact on UX.
+
+Similarly, any known corner cases related to the infrastructure of the cluster can be documented.
+  
+> Note: Users may be allowed to mention such specifications in the test configuration file as mentioned above.
+
+## Dependencies on tools / libraries
+
+- Go
+- shell scripting
+- Docker
+- Sonobuoy
+- Kubectl
+- Linkerd
+- MySQL
+- Redis
+- curl
+- [Ginkgo](https://github.com/onsi/ginkgo) (good-to-have)
+
+
+## Use Cases
+
+This standalone testing tool shall be used for verifying the features of Linkerd involving the data plane, as listed under the "Testing Methodologies for various features" section.
+
+## Goals
+The goal of this project is to :
+1. develop a sample application that can exercise various features of Linkerd, mainly that involve the data plane.
+2. provide an extensive e2e test suite that can make use of this application and validate for conformance.
+
+## Non-goals
+It is a non-goal for this project to provide an application that is expected to be a part of Linkerd application architecture (control / data plane). This project in no way shall directly affect the way Linkerd functions. Unlike the existing integration tests, the e2e tests for conformance in this project are not expected to be a part of the Linkerd CI workflow either. These tests shall run as a stand-alone component such that it can interact with a k8s cluster.
+
+## Deliverables
+
+### Must have (shall be completed by end of GSoC):
+- An e2e test suite that can perform conformance validation for the following features:
+    1. Automatic proxy injection
+    2. `linkerd tap`, `stat`, `routes`, `edges` cmd
+    3. Verifying if `tap` extenstion API server is functional
+    4. Retries and Timeouts
+    5. Verifying if Data Plane proxies are healthy
+    6. Ingress
+- A Sonobuoy plugin for our test suite
+
+- A sample emojivoto app with feature flags that can enable/disable features required for conformance vaidation.
+
+### Good-to-have (if extra time remains, these shall be worked on):
+- The conformance validation testing suite shall also cover the following features :
+    1. Telemetry and metrics
+    2. Cluster security and introspection
+    3. Data plane PSP
+    4. Distributed tracing
+    5. MySQL, Redis
+   
+- A new sample application - MovieChat - that has all the features required from conformance validation perspective
```

Comment:
I recommend that you move these 3 sections to the top of the RFC. Maybe add a _Goals and Deliverables_ parent section between the Phase 1 and Phase 2 sections. This provides your readers with a clear picture of what you will be delivering, before going into the implementation details in the Phase 2 section.